src/federation/observer.rs

```diff
@@ -107,12 +120,10 @@ impl FederationObserver {
 
         for (version, migration) in migration_map.iter() {
             if *version > schema_version {
-                self.connection()
-                    .await?
-                    .transaction()
-                    .await?
-                    .batch_execute(migration)
-                    .await?;
+                let mut conn = self.connection().await?;
+                let transaction = conn.transaction().await?;
+                transaction.batch_execute(migration).await?;
+                transaction.commit().await?;
```

Comment:
I might be misreading the api, but I think we need to commit the transaction once we're finished

Reply:
You are very much correct :see_no_evil: I was doing a lot of manual experimentation with the DB schema, so didn't notice a lot of the problems with automated migrations till I deployed to production and then didn't get around updating the PR after fixing things manually.